src/components/LocationPage/ChartBlock.tsx

```diff
@@ -0,0 +1,145 @@
+import Outcomes from 'components/Outcomes/Outcomes';
+import ShareButtons from 'components/LocationPage/ShareButtons';
+import {
+  ChartRt,
+  ChartPositiveTestRate,
+  ChartICUHeadroom,
+  ChartContactTracing,
+  ChartFutureHospitalization,
+} from 'components/Charts';
+import { Metric, getMetricName } from 'common/metric';
+import { formatUtcDate } from 'common/utils';
+import { generateChartDescription } from 'common/metrics/future_projection';
+
+function Chart(props: { getDatasetData: any; metric: number; data: any }) {
+  return (
+    <Fragment>
+      {props.data && props.metric === Metric.CASE_GROWTH_RATE && (
+        <ChartRt columnData={props.getDatasetData} />
+      )}
+      {props.data && props.metric === Metric.POSITIVE_TESTS && (
+        <ChartPositiveTestRate columnData={props.data} />
+      )}
+      {props.data && props.metric === Metric.HOSPITAL_USAGE && (
+        <ChartICUHeadroom columnData={props.data} />
+      )}
+      {props.data && props.metric === Metric.CONTACT_TRACING && (
+        <ChartContactTracing columnData={props.data} />
+      )}
+    </Fragment>
+  );
+}
+
+function ChartBlock(props: {
+  chartRef: React.RefObject<HTMLDivElement>;
+  isMobile: Boolean;
+  shareButtonProps: { [key: string]: any };
+  projection: Projection;
+  metric: number;
+  datasetName?: any;
+  statusText?: string | null;
+  futureProjectionsDisabled?: Boolean;
+  outcomesProjections?: { [key: string]: any };
+  outcomesColors?: string[];
+  noInterventionProjection: Projection;
+  projections: Projections;
+  data?: any;
+}) {
+  const showBetaTag =
+    props.metric === Metric.HOSPITAL_USAGE ||
+    props.metric === Metric.CONTACT_TRACING;
+
+  return (
+    <Fragment>
+      {props.metric !== Metric.FUTURE_PROJECTIONS && (
+        <Fragment>
+          <ChartHeaderWrapper>
+            <ChartHeader ref={props.chartRef}>
+              {getMetricName(props.metric)}
+              {showBetaTag && <BetaTag>Beta</BetaTag>}
+            </ChartHeader>
+            {!props.isMobile && props.data && (
+              <ShareButtons
+                chartIdentifier={props.metric}
+                {...props.shareButtonProps}
+              />
+            )}
+          </ChartHeaderWrapper>
+          <ChartLocationName>{props.projection.locationName}</ChartLocationName>
+          <ChartDescription>{props.statusText}</ChartDescription>
+          {props.isMobile && props.data && (
+            <ShareButtons
+              chartIdentifier={props.metric}
+              {...props.shareButtonProps}
+            />
+          )}
+          {props.data && (
+            <>
+              <Chart
+                data={props.data}
+                metric={props.metric}
+                getDatasetData={props.projection.getDataset(props.datasetName)}
+              />
+              <Disclaimer metricName={props.metric} />
+            </>
+          )}
+        </Fragment>
+      )}
+      {props.metric === Metric.FUTURE_PROJECTIONS && (
```

Comment:
I kept the future projections chart separate of the mapped charts because the conditions/content was a bit different